snap/validate.go

```diff
@@ -394,25 +394,40 @@ func validateAppSocket(socket *SocketInfo) error {
 	return validateSocketAddr(socket, "listen-stream", socket.ListenStream)
 }
 
+func injectFakeExternalApps(apps map[string]*AppInfo) {
+	for _, app := range apps {
+		var beforeAfter []string
+		beforeAfter = append(beforeAfter, app.Before...)
+		beforeAfter = append(beforeAfter, app.After...)
+		for _, other := range beforeAfter {
+			if !strings.HasPrefix(other, "external:") {
+				continue
+			}
+			apps[other] = &AppInfo{
```

Comment:
I think you need to copy the map otherwise you may unintentionally modify `info.Apps` which is originally passed to `validateAppOrderCycles`.